webapp/channels/src/components/suggestion/search_date_suggestion/search_date_suggestion.tsx

```diff
@@ -2,84 +2,72 @@
-import React from 'react';
+import React, {useCallback, useEffect, useState} from 'react';
 import {DayPicker} from 'react-day-picker';
 
-import Constants from 'utils/constants';
-import * as Keyboard from 'utils/keyboard';
-import * as Utils from 'utils/utils';
-
+import Constants from '../../../utils/constants';
+import * as Keyboard from '../../../utils/keyboard';
+import * as Utils from '../../../utils/utils';
 import type {SuggestionProps} from '../suggestion';
 
-import 'react-day-picker/dist/style.css';
-
 type Props = SuggestionProps<never> & {
     currentDate?: Date;
     handleEscape?: () => void;
     locale: string;
     preventClose?: () => void;
 }
 
-export default class SearchDateSuggestion extends React.PureComponent<Props> {
-    private loadedLocales: Record<string, Locale> = {};
-
-    state = {
-        datePickerFocused: false,
-    };
+const SearchDateSuggestion = ({currentDate, handleEscape, locale, preventClose, matchedPretext, onClick}: Props) => {
+    let loadedLocales: Record<string, Locale> = {};
+    loadedLocales = Utils.getDatePickerLocalesForDateFns(locale, loadedLocales);
+    const [datePickerFocused, setDatePickerFocused] = useState(false);
 
-    handleDayClick = (day: Date) => {
+    const handleDayClick = useCallback((day: Date) => {
         const dayString = new Date(Date.UTC(day.getFullYear(), day.getMonth(), day.getDate())).toISOString().split('T')[0];
-        this.props.onClick(dayString, this.props.matchedPretext);
-    };
-
-    handleKeyDown = (e: KeyboardEvent) => {
-        if (Keyboard.isKeyPressed(e, Constants.KeyCodes.DOWN) && document.activeElement?.id === 'searchBox') {
-            this.setState({datePickerFocused: true});
-        } else if (Keyboard.isKeyPressed(e, Constants.KeyCodes.ESCAPE)) {
-            this.props.handleEscape?.();
-        }
-    };
-
-    componentDidMount() {
-        document.addEventListener('keydown', this.handleKeyDown);
-    }
-
-    componentWillUnmount() {
-        document.removeEventListener('keydown', this.handleKeyDown);
-    }
-
-    iconLeft = () => {
+        onClick(dayString, matchedPretext);
+    }, [onClick, matchedPretext]);
+
+    useEffect(() => {
+        const handleKeyDown = (e: KeyboardEvent) => {
+            if (Keyboard.isKeyPressed(e, Constants.KeyCodes.DOWN) && document.activeElement?.id === 'searchBox') {
+                setDatePickerFocused(true);
+            } else if (Keyboard.isKeyPressed(e, Constants.KeyCodes.ESCAPE)) {
+                handleEscape?.();
+            }
+        };
+        document.addEventListener('keydown', handleKeyDown);
+        return () => {
+            document.removeEventListener('keydown', handleKeyDown);
+        };
+    }, []);
+
+    const iconLeft = () => {
         return (
             <i className='icon icon-chevron-left'/>
         );
     };
 
-    iconRight = () => {
+    const iconRight = () => {
         return (
             <i className='icon icon-chevron-right'/>
         );
     };
```

Comment:
These don't depend on the props/state, so they can be defined outside of the component. On top of that, they can be considered components (function components) so make the names uppercase (i.e. `IconLeft`)

Reply:
Done